substratest/assets.py

```diff
@@ -396,14 +387,13 @@ class Meta:
         }
 
 
-@dataclasses.dataclass
-class ComputePlan(_Asset, _ComputePlanFutureMixin):
-    compute_plan_id: str
+class ComputePlan(_ComputePlanFutureAsset):
+    compute_plan_id: str = None
     status: str
-    traintuple_keys: typing.List[str]
-    composite_traintuple_keys: typing.List[str]
-    aggregatetuple_keys: typing.List[str]
-    testtuple_keys: typing.List[str]
+    traintuple_keys: typing.List[str] = None
+    composite_traintuple_keys: typing.List[str] = None
+    aggregatetuple_keys: typing.List[str] = None
+    testtuple_keys: typing.List[str] = None
     tag: str
 
     def __post_init__(self):
```

Comment:
Could we get rid of the __post_init__ method?
It would be good to ensure the default value of the `<tuple>_keys` fields is an empty list and not a None value.

Reply:
I'll try to, but I'm not sure that it is possible

Reply:
When I try to replace `None` by `[]`, I get this error:

```
E           pydantic.error_wrappers.ValidationError: 3 validation errors for ComputePlan
E           composite_traintuple_keys
E             none is not an allowed value (type=type_error.none.not_allowed)
E           aggregatetuple_keys
E             none is not an allowed value (type=type_error.none.not_allowed)
E           testtuple_keys
E             none is not an allowed value (type=type_error.none.not_allowed)
```

Reply:
The issue is that the chaincode returns null instead of an empty list. I'm afraid we can't solve it here.
As for the default `[]` value, aren't we going to run into this classic issue?
```
In [1]: def foo(l=[]):
   ...:     l.append('foo')
   ...:     return l

In [2]: foo()
Out[2]: ['foo']

In [3]: foo()
Out[3]: ['foo', 'foo']
```

Reply:
Good point, thanks @jmorel. So I think it should be done manually in the constructor function (something similar to the dataclass `__post_init__` function).